### ISSUE TITLE
Chainer model problem?

### ISSUE THREAD

**Issue by mjiansun** [pfnet/chainer#2444]
**First of all, I do 3D convolution.Data is cuda.cupy.array.**
**In my model code,** `h2 = F.relu(self.bnorm2_8(h2, test=not train))` **h2 shape is (20,1,3,3,3).I want to enlarge it to (20,1,9,9,9).So i did the following:**
h2.data = h2.data.repeat(3,2)
h2.data = h2.data.repeat(3,3)
h2.data = h2.data.repeat(3,4)

**Next,i want to concate two same size tensor.** `c = cuda.cupy.concatenate((h1.data,h2.data),axis=1)`

**So i have a question:
1)If I had the above operation, will not affect Variable's gradient?**

**Comment by delta2323** [comment]
Hi. As repeat operations are not registered to the computational graph, the computation of gradient will be wrong (I guess shape mismatch between `Variable.data` and `Variable.grad` will be caused).

Repeat operations should be done at `chainer.Function` level. You can do it with `chainer.functions.tile` (e.g. `F.tile(h2, (3, 3, 3))`)

**Comment by mjiansun** [comment]
**When i use my default parameter:**
h2.data = h2.data.repeat(3,2)
h2.data = h2.data.repeat(3,3)
h2.data = h2.data.repeat(3,4)
c = cuda.cupy.concatenate((h1.data,h2.data),axis=1)

**My net is running successfully.So What am I going to do？**

**Comment by delta2323** [comment]
If you do not have to compute gradients of `c` (i.e. execute backward propagation from `c`). That works fine. But if you do that, you should manipulate at `Variable` level without extracting `Variable.data` attribute.

**Comment by mjiansun** [comment]
@delta2323 Thank you very much for your answer.In my model, I change my code to 
`c = cuda.cupy.concatenate((h1.data,h2.data),axis=1)`
`c=Variable(c)`
So, I guess it might be wrong.

**Comment by delta2323** [comment]
Chainer has its counter part [`chainer.functions.concat`](http://docs.chainer.org/en/latest/reference/functions.html#concat). So, the resulting code would be:

```
h2 = F.relu(self.bnorm2_8(h2, test=not train))
h2 = F.tile(h2, (3, 3, 3))
c = F.concat((h1, h2), axis=1)  # or equivalently, c = F.hstack((h1, h2))
```

**Comment by mjiansun** [comment]
Thank you very much!
